controllers/configurationpolicy_controller.go

```diff
@@ -262,11 +262,7 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(
 	}
 
 	// initialize the RelatedObjects for this Configuration Policy
-	oldRelated := []policyv1.RelatedObject{}
-
-	for i := range plc.Status.RelatedObjects {
-		oldRelated = append(oldRelated, plc.Status.RelatedObjects[i])
-	}
+	oldRelated := plc.Status.RelatedObjects
```

Comment:
Are you sure this is safe? This is different behavior than what was previously there since even though you are assigning this to a variable, it contains the same slice `plc.Status.RelatedObjects`. The original behavior essentially created a copy.

See this playground for an example:
https://go.dev/play/p/VkDtSmWJ2ul

Reply:
Yeah, you're right, this is a rough edge I've run into before - slices are pointers to backing arrays, so you can get into weird stuff...

The tests passed, which made me more confident, but you're right that it could still run into trouble.

Reply:
I'm good with it if the Config Policy controller e2e tests from the policy framework also pass.